lang/src/org/partiql/lang/eval/time/Time.kt

```diff
@@ -180,4 +181,15 @@ data class Time private constructor(val localTime: LocalTime, val precision: Int
     override fun toString(): String =
         localTime.format(DateTimeFormatter.ofPattern(formatterPattern())) +
             (zoneOffset?.getOffsetHHmm() ?: "")
+
+    override fun compareTo(other: Time): Int {
+        return when {
+            // When the zone offsets are not null for both the operands, compare OffsetTime i.e. LocalTime with ZoneOffset
+            this.zoneOffset != null && other.zoneOffset != null -> this.offsetTime!!.compareTo(other.offsetTime)
+            // When the zone offsets are null for both the operands, compare just LocalTime
+            this.zoneOffset == null && other.zoneOffset == null -> this.localTime.compareTo(other.localTime)
+            // When one of the times is `time with time zone` and other is just `time` (without time zone), then they are incomparable.
+            else -> errNoContext("Cannot compare values: $this, $other", internal = false)
+        }
+    }
```

Comment:
So I believe that this is good for `<`/`>`/`<=`/`>=`/`=` operators, but not good for the natural comparator for `Time` values. 

Put another way, for the natural order it would be okay with all `Time` without time zone to compare before or after the `Time` with time zones, but I don't think it should be an error, otherwise `GROUP BY` and `ORDER BY` won't work.

Reply:
Got it. Thanks. I have removed the `Comparable` interface from which the `Time` class was derived and have included `naturalOrderCompareTo` function to include the natural ordering logic as well. The overridden `compareTo` method, in my opinion, should still throw an error when the `Time` instances are compared. However, since it is not used anywhere currently, I have decided to remove it for now.